report/4.Solution/ParserImplementation.tex

```diff
@@ -115,7 +118,83 @@ \subsection{Implementing a recursive decent parser}
 The method then checks whether the new current symbol is an equals-symbol - if that's the case the declaration is followed by an expression - which means it's an declaration with an assignment otherwise the expression is set to null. 
 The method then returns a Declaration, which is a subclass of AstNode.  
 
+A more complex parsing routine is when an expression is to be parsed, since an expression in HCL may be either a function-call or any kind of value.
+This means that there are a lot of options for the parser when parsing expressions.
+The parser first tries to parse a potential function-call using the \textbf{\textit{parsePotentialFunctionCall}} method.
+
+\begin{lstlisting}[language=Kotlin,label=lis:parsePotCall,caption=A simplified version of the parsePotentialFunctionCall method from the parser.]
+private fun parsePotentialFunctionCall(expression: AstExpression? = null): AstExpression =
+    when (current.token) {
+        is Token.SpecialChar.BlockStart -> {
+            val upcomingId = getUpcomingIdentifierNameForLambda()
+            if (upcomingId != null) {
+                val symbol = retrieveSymbol(upcomingId)
+                val lambdaParameter = getLambdaParameter(symbol.functions, 0)
+                parsePotentialFunctionCall(lambdaParameter)
+            } else {
+                val lambda = parseLambdaBody(emptyList())
+                AstNode.Command.Expression.LambdaExpression(listOf(), lambda.type, lambda.lambdaBody)
+            }
+        is Token.Identifier -> {
+            val token = accept<Token.Identifier>()
+            val symbol = retrieveSymbol(token.value)
+            if (symbol.isFunctions) {
+        	    ...parse needed arguments...
+        	    AstNode.Command.Expression.FunctionCall(
+        	    AstNode.Command.Expression.Value.Identifier(token.value),
+        	    listOf(expression) + secondaryArguments
+            } else expression
+        } else -> expression
+\end{lstlisting}
+
+As seen in snippet \ref{lis:parsePotCall}, the method first determines if the current token is a lambda expression, signified by a \textit{blockStart-token} (3).
+Should it be a lambda expression, the parser determines whether or not there is a functions call within the lambda that needs to be parsed as well (7).
+If there is not a function-call within the lambda, the lambda expression itself is returned from the method (10). 
```

Comment:
Change "Lambda" -> "Lambda expression" 
OR
Change "Lambda expression" -> "Lambda"

Reply:
In the whole section offcause. Just to be consistent :)